src/luarocks/type_check.lua

```diff
@@ -58,6 +58,7 @@ local rockspec_types = {
       dir = string_1,
       tag = string_1,
       branch = string_1,
+      revision = string_1,
```

Comment:
You should use `revision = { _type = "string", _version = "3.0" }` here (or you can even create a `string_3` variable)